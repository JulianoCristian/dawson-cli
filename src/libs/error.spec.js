import test from 'ava';
import createError from './error';

test('creates an error', t => {
  const err = createError({
    kind: '3452345',
    reason: '53656564',
    detailedReason: '2893475923847593',
    solution: '985u4986'
  });
  t.true(err.isDawsonError);
  t.snapshot(err.toFormattedString());
});

test('creates an error with default params', t => {
  const err = createError({});
  t.true(err.isDawsonError);
  t.snapshot(err.toFormattedString());
});
