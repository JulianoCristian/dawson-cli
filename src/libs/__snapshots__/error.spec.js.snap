exports[`creates an error 1`] = `
"
   [41m[1mExecution Error[22m[49m
   [31m[1m3452345[22m[39m

   [31m[1mPossible cause[22m[39m
   [31m53656564[39m

   2893475923847593

   [32m[1mPossible solution[22m[39m
   985u4986

   [90mIf you believe this is not an expected behaviour or that this error message should be improved, open an issue: https://github.com/dawson-org/dawson-cli/issues[39m"
`;

exports[`creates an error with default params 1`] = `
"
   [41m[1mExecution Error[22m[49m

   [90mIf you believe this is not an expected behaviour or that this error message should be improved, open an issue: https://github.com/dawson-org/dawson-cli/issues[39m"
`;
